[je-nnyt] iP

je-nnyt opened the pull request.

je-nnyt committed:
Changed name to koya, add greetings and exit

je-nnyt committed:
Level-0

je-nnyt committed:
Merge branch 'master' of https://github.com/je-nnyt/ip

je-nnyt committed:
echos commands entered by user

and exits when user enters "bye" in the command

je-nnyt committed:
add ability to store information entered by the user and to list them back in order when requested

je-nnyt committed:
added the class Task and the ability to mark tasks as done by displaying a status icon

je-nnyt committed:
Tweaked code and IDE settings to follow coding standards

Modified the if-else statements and variables name

je-nnyt committed:
Add Todo, Deadline, Event subclass
Tested Todo in main

je-nnyt committed:
Added and tested methods to deal with deadline objects in main

je-nnyt committed:
Added and tested methods to deal with Event objects in main

je-nnyt committed:
Extracted methods: confirmMarkDone, listTasks, confirmAddTask
Set taskCount as a global variable to fix taskCount not incrementing after the task has been added to list
Add comments, reformatting code

je-nnyt committed:
Refactor project to follow Java coding standards

- Move each class to its own Java file
- Rename class 'Todo' to 'ToDo'
- Create package 'ip.task' to organize Task and its subclasses (ToDo, Deadline, Event)

LEESY02 commented on src/main/java/Koya.java:
```diff
@@ -0,0 +1,94 @@
+
+        String input;
+        Scanner in = new Scanner(System.in);
+        System.out.println("What can I do for you?");
```
Perhaps this print message and the 1st print message could be combined into one, and abstracted to a separate private method for greater readability.

private static void main(String[] args) {
printIntroMessage();
........
}

private static void printIntroMessage() {
sout("This is an intro message";
}

Maybe same could be applied for all other print messages

yangyi-zhu replied:
I agree with this for it would reduce the number of lines required for this segment. Also, grouping similar lines of code would improve the overall presentation.

LEESY02 commented on src/main/java/Koya.java:
```diff
@@ -0,0 +1,94 @@
+
+            if (input.equals("bye")) {
+                break;
+            } else if (input.equals("list")) {
```
Maybe a switch statement will be easier to read? 

I believe you could do in.next() to just check the 1st word or split the whole line into an array using " " (whitespace) as a separator, and use that word in the case/switch block

LEESY02 commented on src/main/java/Koya.java:
```diff
@@ -0,0 +1,94 @@
+                int fromDividerPosition = input.indexOf(" /from ");
+                int toDividerPosition = input.indexOf(" /to ");
+
+                String description = input.substring(6, fromDividerPosition);
```
I think there would be other ways of implementing this instead of using magic numbers, such as the methods I mentioned in my previous comment

LEESY02 commented on src/main/java/Koya.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    private static void confirmMarkDone(Task list) {
+        System.out.println("Nice! I've marked this as done:");
```
nit picking but shouldn't the name of the Task variable be task? Writing it as list is kind of confusing, as it is just 1 variable, not a Task[] list

LEESY02 commented on src/main/java/ip/task/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+package ip.task;
```
nice implementation of child class :D

LEESY02 reviewed (commented):
Seems great, just had a few nits to pick :D

hyunjinkim1112 commented on src/main/java/Koya.java:
```diff
@@ -124,7 +40,7 @@ public static void main(String[] args) {
             } else if (input.startsWith("todo")) {
```
I liked that you used .startsWith () function to parse the command input. It makes the code simple.

hyunjinkim1112 commented on src/main/java/Koya.java:
```diff
@@ -1,93 +1,9 @@
+import ip.task.Deadline;
+import ip.task.Event;
+import ip.task.Task;
+import ip.task.ToDo;
```
Good attempt to make a java package and import !

yangyi-zhu replied:
Great to modularize and use imports instead of putting everything together! Greatly improves the readability!

hyunjinkim1112 commented on src/main/java/Koya.java:
```diff
@@ -124,7 +40,7 @@ public static void main(String[] args) {
+                list[taskCount] = new ToDo(description);
                 confirmAddTask(list);
 
             } else if (input.startsWith("deadline")) {
```
You may consider making the class that process the input like .processInput() which will make your main method shorter and more concise.

hyunjinkim1112 commented on src/main/java/Koya.java:
```diff
@@ -124,7 +40,7 @@ public static void main(String[] args) {
 
-                list[taskCount] = new Todo(description);
+                list[taskCount] = new ToDo(description);
                 confirmAddTask(list);
```
You may want to increase the taskCount++ here because confirmAddTask(list)'s functionality is to print the added task to confirm (not really adding a task; it has been added above).

hyunjinkim1112 reviewed (commented):
Nice and readable code! Good work!

yangyi-zhu commented on src/main/java/Koya.java:
```diff
@@ -90,83 +90,89 @@ public String toString() {
+                list[taskCount] = new Event(description, from, to);
+                confirmAddTask(list);
+            } else {
+                System.out.println("Action unsupported. Try again.");
```
Considering possible alternatives is a great practice! However, I would suggest defaulting to a type if no expected prefix was detected instead.

yangyi-zhu commented on src/main/java/ip/task/Task.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    public String getTaskTypeIcon() {
+        return " ";//leave empty, overridden by subclass
```
I like the approach incorporating inheritance, makes it much easier to navigate from a developer/reviewer standpoint.
I think this line in particular can be further simplified by employing the abstract modifier.

yangyi-zhu commented on src/main/java/ip/task/Deadline.java:
```diff
@@ -0,0 +1,22 @@
+
+    @Override
+    public String toString() {
+        return super.toString() + " (by: " + by + ")";
```
Excellent example of how overriding methods can expand upon their equivalents in their parent class :D

yangyi-zhu reviewed (commented):
LGTM overall! Had to spend some time to nit out a few minor ways I think the presentation could improve. Be sure to keep up with the work! 👯

je-nnyt committed:
Adjust code based on PR's review feedback
- Extract the printIntroMessage() method
- Add comments for magic indexing numbers

je-nnyt committed:
Add class KoyaException

je-nnyt committed:
Add try-catch block to handle exceptions
- Try-catch block handles the empty to do description and the unknown input

je-nnyt committed:
Add Package

je-nnyt committed:
Reformat after adding packages

je-nnyt committed:
Replace Task[] with ArrayList<Task> to store the tasks and modify methods accordingly

je-nnyt committed:
Add removeTask method

je-nnyt committed:
Add appendtoFile method
- check if folder and file exists
- load task from txt file to list array
- only toDo task can be added to list

je-nnyt committed:
Add ability to load Deadline task from txt file back to list

je-nnyt committed:
Add ability to load Deadline and Event task into the list array

je-nnyt committed:
reformat the code

je-nnyt committed:
Resolve conflicts after merging  'branch-Level-6' and branch-Level-7'

thomasjlalba commented on src/main/java/koya/Koya.java:
```diff
@@ -0,0 +1,190 @@
+
+        String input;
+        Scanner in = new Scanner(System.in);
+        printIntroMessage();
```
Good job on abstracting to smaller methods!

thomasjlalba commented on src/main/java/koya/Koya.java:
```diff
@@ -0,0 +1,190 @@
+                    list.get(taskIndex).markAsDone();
+                    confirmMarkDone(list.get(taskIndex));
+
+                } else if (input.startsWith("todo")) {
+
+                    if (input.length() <= TODO_CHAR_COUNT) {
+                        throw new KoyaException("OOH OH! The description of a todo cannot be left empty...");
+                    }
+
+                    String description = input.substring(5);
+                    list.add(new ToDo(description));
+                    try {
+                        appendToFile(FILE_PATH, list.get(taskCount));
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                    confirmAddTask(list);
+
+                } else if (input.startsWith("deadline")) {
```
Maybe you can avoid deep nesting by [refactoring to its own method](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard)

thomasjlalba commented on src/main/java/koya/Koya.java:
```diff
@@ -0,0 +1,190 @@
+                    //Parsing index to obtain description and by
+                    int dividerPosition = input.indexOf(" /by ");
+
+                    String description = input.substring(9, dividerPosition); // 9: 8 for deadline + 1 for the space
+                    String by = input.substring(dividerPosition + 5); // 5 for the number of characters in " /by "
```
Maybe you can [avoid magic literals](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers) to make it more readable

thomasjlalba commented on src/main/java/koya/Koya.java:
```diff
@@ -0,0 +1,190 @@
+                } else if (input.startsWith("delete")) {
+                    //indexes to obtain number next to delete
+                    int spaceIndex = input.indexOf(" ");
+                    int taskIndex = Integer.parseInt(input.substring(spaceIndex + 1)) - 1;
```
Maybe you can make it more readable by splitting the line to 2 lines so that it is easier to understand

thomasjlalba commented on src/main/java/koya/Koya.java:
```diff
@@ -0,0 +1,190 @@
+            throw new RuntimeException(e);
+        }
+
+        while (true) {
```
Maybe you can look to separate this out to its own method to shorten the main method

thomasjlalba reviewed (commented):
Good job thus far! Just some suggestions to make your code look neater and improve the quality

je-nnyt committed:
Fix delete Task bug
- txt file now updates when task is deleted

je-nnyt committed:
Fix bug display of mark as done in 'list'
- add trim to parts[1] to remove trailing spaces
- list method now properly displays [X] when loading data from file

je-nnyt committed:
Extract class UI

je-nnyt committed:
Extract class Storage

je-nnyt committed:
Extract class TaskList

je-nnyt committed:
Extract class Parser

je-nnyt committed:
Add function findMatchingTask
- Fix bug of printing null for 'description', 'by', 'to', 'from' for Event and Deadline tasks

je-nnyt committed:
Add JavaDoc comment for Storage, TaskList, Ui, Deadline, Event, ToDo

je-nnyt committed:
Merge pull request #1 from je-nnyt/branch-Level-9

Add function findMatchingTask

je-nnyt committed:
Merge branch 'branch-A-JavaDoc'

je-nnyt committed:
Add a draft User Guide
- contains information on the important features
- missing the quick start and formatting

je-nnyt committed:
Fix README.md not showing up properly
- fix display bug of event

je-nnyt committed:
Fix find method to list all task with given keyword
- user no longer needs to input the exact description
- method now displays all tasks with the same keyword

je-nnyt committed:
Update README.md with a Quick Start section

je-nnyt committed:
Add trim function to methods to remove trailing spaces

je-nnyt committed:
Refactor README and add more detail for usage of delete and task

Rewording of sentences to make the user guide more comprehensive. Add a note section for delete and mark.

je-nnyt committed:
Refactor main class Koya and add a catch Exception to the main method

Main method is unable to catch exception of type Exception and overall readability of the main can be improved

je-nnyt committed:
Refactoring of code

je-nnyt committed:
Refactor code to improve code quality

Modify the example of the find section of the README to make it more detailed.
Modify the throw message to be more specific which may help the debugging process.

je-nnyt committed:
Improve readability and visuals of User Guide

je-nnyt committed:
Update the README.md to put back in place the initial Duke project template

I initially wrote on the README.md not found in docs, which removed the Duke project template

je-nnyt committed:
Add JavaDoc for public methods

je-nnyt committed:
Add JavaDoc comments to public methods

je-nnyt committed:
Add try/catch block for more exception handling

Program ends if the user leaves event, deadline, delete and find empty.

je-nnyt committed:
Add and update JavaDoc comments

Missing JavaDoc comments for public methods.

je-nnyt committed:
Add a dashedLine method to improve UI of program

Cluttered UI makes it hard for the user to read program.

je-nnyt committed:
Update KoyaException message for better UI clarity

je-nnyt committed:
Add dashedLine() to main method

Improve cluttered UI.

je-nnyt committed:
Improve code quality

Remove block of comments and improve header comments.

je-nnyt committed:
Refactor code

je-nnyt committed:
Update JavaDoc header comments for clarity
